tests/php/Functional/WC/Mollie_WC_Plugin_Test.php

```diff
@@ -264,4 +265,67 @@ protected function setUp()
 
         when('filemtime')->justReturn($this->fileMTime);
     }
+    /* -----------------------------------------------------------------
+       onMollieReturn Tests
+       -------------------------------------------------------------- */
+
+    /**
+     * Given onMollieReturn is called
+     * when order and gateway are valid
+     * then we get a redirectUrl
+     * @test
+     */
+    public function onMollieReturn_redirectUrlCorrect(){
+        $redirectUrl= 'https://website.org/language/order-received/4927/?key=wc_order_zUHGhd2f4fGc9';
+        /*
+         * Setup Stubs to mock the external dependencies and its methods
+         */
+        $dataHelper = $this->buildTesteeMock(
```

Comment:
`buildTesteeMock` it's just for SUT/Testee I would use Phpunit api here and everywhere you need a mock for something which is not the SUT.

Btw this test sounds more unit than functional.

Do you think we could make it smaller? Would be difficult to understsand this code in the near future.